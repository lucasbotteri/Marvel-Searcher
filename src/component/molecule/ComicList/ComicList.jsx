import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import ComicItem from "../../molecule/ComicItem";

export const ComicListWrapper = styled.ul`
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  margin: 0;
  padding: 0;
`;

export const NonStyledLi = styled.li`
  list-style: none;
`;

const ComicList = ({ comics }) => {
  return (
    <ComicListWrapper>
      {comics.map((c) => (
        <NonStyledLi key={c.id}>
          <ComicItem comic={c} />
        </NonStyledLi>
      ))}
    </ComicListWrapper>
  );
};

ComicList.propTypes = {
  comics: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      title: PropTypes.string,
      description: PropTypes.string,
      thumbnail: PropTypes.string,
      isFaved: PropTypes.bool,
    })
  ).isRequired,
};

export default ComicList;
