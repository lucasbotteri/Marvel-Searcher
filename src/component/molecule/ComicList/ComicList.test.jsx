import React from "react";
import { shallow } from "enzyme";
import ComicList, { NonStyledLi } from "./ComicList";

const comics = [
  {
    id: 735,
    title: "Amazing Fantasy (2004) #1",
    description:
      "She's fierce, she's sassy...she sticks to walls! AMAZING FANTASY #1 is where it all begins: Meet an all-new, all-different heroine! Fiona Avery (AMAZING SPIDER-MAN) and Mark Brooks (MARVEL AGE SPIDER-MAN) launch a surprising series of teen-friendly adventures set in the current Marvel Universe!",
    thumbnail:
      "http://i.annihil.us/u/prod/marvel/i/mg/9/80/56cb7746bd44a/standard_medium.jpg",
  },
  {
    id: 529,
    title: "Amazing Fantasy (2004) #2",
    description:
      "Sugar and spice and everything nice - that's what this little girl is totally NOT made of!  Marvel's scrappy new hero meets a tall, dark, and handsome stranger who leaves her with more questions than answers.  Who is this guy, anyway? And why has he taken such an interest in her?",
    thumbnail:
      "http://i.annihil.us/u/prod/marvel/i/mg/6/30/56cb6bb538602/standard_medium.jpg",
  },
];

describe("ComicList", () => {
  it("Should render without crashing", () => {
    const wrapper = shallow(<ComicList comics={comics} />);
    expect(wrapper).toBeTruthy();
  });

  it("Should render two items", () => {
    const wrapper = shallow(<ComicList comics={comics} />);
    expect(wrapper.find(NonStyledLi)).toHaveLength(2);
  });

  it("Should match snapshot", () => {
    const wrapper = shallow(<ComicList comics={comics} />);
    expect(wrapper).toMatchSnapshot();
  });
});
