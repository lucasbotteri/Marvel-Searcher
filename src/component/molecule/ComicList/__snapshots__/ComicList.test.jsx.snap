// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComicList Should match snapshot 1`] = `
<styled.ul>
  <styled.li
    key="735"
  >
    <ComicItem
      comic={
        Object {
          "description": "She's fierce, she's sassy...she sticks to walls! AMAZING FANTASY #1 is where it all begins: Meet an all-new, all-different heroine! Fiona Avery (AMAZING SPIDER-MAN) and Mark Brooks (MARVEL AGE SPIDER-MAN) launch a surprising series of teen-friendly adventures set in the current Marvel Universe!",
          "id": 735,
          "thumbnail": "http://i.annihil.us/u/prod/marvel/i/mg/9/80/56cb7746bd44a/standard_medium.jpg",
          "title": "Amazing Fantasy (2004) #1",
        }
      }
    />
  </styled.li>
  <styled.li
    key="529"
  >
    <ComicItem
      comic={
        Object {
          "description": "Sugar and spice and everything nice - that's what this little girl is totally NOT made of!  Marvel's scrappy new hero meets a tall, dark, and handsome stranger who leaves her with more questions than answers.  Who is this guy, anyway? And why has he taken such an interest in her?",
          "id": 529,
          "thumbnail": "http://i.annihil.us/u/prod/marvel/i/mg/6/30/56cb6bb538602/standard_medium.jpg",
          "title": "Amazing Fantasy (2004) #2",
        }
      }
    />
  </styled.li>
</styled.ul>
`;
